lib/postgres/postgres-pr/message.rb

```diff
@@ -46,7 +46,11 @@ def self.read(stream, startup=false)
     type = stream.read_exactly_n_bytes(1) unless startup
     length = stream.read_exactly_n_bytes(4).to_s.unpack('N').first  # FIXME: length should be signed, not unsigned
 
-    raise ParseError if (length.nil? || length < 4)
+    if length.nil?
+      raise EOFError
```

Comment:
Looks like this behavior was lost in this custom code here:

https://github.com/rapid7/metasploit-framework/blame/ceb65d7dc0feda7930c9ac2664818ce1efc2743a/lib/postgres/postgres-pr/message.rb#L12-L26

I think this code belongs in the monkey patch, but since there's only one place using it - so I'm also okay with that

Reply:
Let's move it to the patch